In [1]:
import pandas as pd
import numpy as np

In [13]:
df = pd.read_excel("/home/jupyter/src/dataviz/podcastcount/2LMarketBasket.xlsx")

In [14]:
df.shape

(10931, 44)

In [15]:
list(df)

['NPI',
 'HCP ID',
 'First Name',
 'Last Name',
 'Specialty',
 'Sales Tier',
 'Facility Tier',
 '2L+ TKI Target',
 'LTD Reach - All Reach Methods',
 'LTD Reach - Live',
 'April Marketing Campaign - Email Open Count',
 'Total GIST Patients',
 'Total Treated GIST Patients',
 'Total 3L+ GIST Patients',
 'Total Ayva GIST Patients',
 'Komodo Clinical Experience Score',
 'Komodo Clinical Volume Score',
 'GLEEVEC Units - 52 Week',
 'GLEEVEC Units - 26 Week',
 'Sutent Units - 52 Week',
 'Sutent Units - 26 Week',
 'Stivarga Units - 52 Week',
 'Stivarga Units - 26 Week',
 'Ayva Units - 52 Week',
 'Ayva Units - 26 Week',
 'Other TKI Units - 52 Week',
 'Other TKI Units - 26 Week',
 '2L+ TKI Units - 52 Week',
 '2L+ TKI Units - 26 Week',
 'Field Reported 2L+',
 'LTD QINLOCK Rx (HCP)',
 'Region',
 'Territory',
 'Facility ID',
 'Facility Name',
 'Facility Address',
 'Facility Address 2',
 'Facility City',
 'Facility State',
 'Facility Zip',
 'Facility Phone',
 'Facility Fax',
 'Network/Parent',
 'Faci

In [16]:
pd.pivot_table(df,index=["2L+ TKI Target"])

2L+ TKI Units - 26 Week  2L+ TKI Units - 52 Week  \
2L+ TKI Target                                                     
No                             0.132070                 0.251601   
Yes                            1.921801                 3.693523   

                April Marketing Campaign - Email Open Count  \
2L+ TKI Target                                                
No                                                 3.212101   
Yes                                                3.133333   

                Ayva Units - 26 Week  Ayva Units - 52 Week  Facility Zip  \
2L+ TKI Target                                                             
No                          0.001067              0.001067  47067.107192   
Yes                         0.032385              0.052133  53843.667114   

                Field Reported 2L+  GLEEVEC Units - 26 Week  \
2L+ TKI Target                                                
No                        0.000000                 2.151547   
Yes                       1.435013                 5.001580   

                GLEEVEC Units - 52 Week  Komodo Clinical Experience Score  \
2L+ TKI Target                                                              
No                             4.215582                          2.165924   
Yes                            9.979463                          2.892233   

                ...  Other TKI Units - 26 Week  Other TKI Units - 52 Week  \
2L+ TKI Target  ...                                                         
No              ...                   0.053629                   0.116329   
Yes             ...                   0.668246                   1.307267   

                Stivarga Units - 26 Week  Stivarga Units - 52 Week  \
2L+ TKI Target                                                       
No                              0.012540                  0.023746   
Yes                             0.301738                  0.571090   

                Sutent Units - 26 Week  Sutent Units - 52 Week  \
2L+ TKI Target                                                   
No                            0.064835                0.110459   
Yes                           0.919431                1.763033   

                Total 3L+ GIST Patients  Total Ayva GIST Patients  \
2L+ TKI Target                                                      
No                             1.025000                  1.000000   
Yes                            1.137778                  1.117647   

                Total GIST Patients  Total Treated GIST Patients  
2L+ TKI Target                                                    
No                         2.535219                     1.453972  
Yes                        4.762243                     2.693223  

[2 rows x 23 columns]

In [17]:
pd.pivot_table(df,index=["2L+ TKI Target","Region"])

2L+ TKI Units - 26 Week  2L+ TKI Units - 52 Week  \
2L+ TKI Target Region                                                          
No             Central                     0.132107                 0.244147   
               Great Lakes                 0.209841                 0.399421   
               Northeast                   0.023739                 0.051929   
               South                       0.161572                 0.269287   
               Southeast                   0.082982                 0.205345   
               West                        0.219638                 0.400517   
Yes            Central                     2.041885                 3.921466   
               Great Lakes                 2.004902                 4.004902   
               Northeast                   2.657754                 4.791444   
               South                       1.932990                 3.886598   
               Southeast                   1.862559                 3.530806   
               West                        1.322581                 2.562724   

                            April Marketing Campaign - Email Open Count  \
2L+ TKI Target Region                                                     
No             Central                                         3.540000   
               Great Lakes                                     3.204724   
               Northeast                                       3.120846   
               South                                           3.047210   
               Southeast                                       3.040590   
               West                                            3.388889   
Yes            Central                                         3.181818   
               Great Lakes                                     3.638298   
               Northeast                                       2.808511   
               South                                           3.409091   
               Southeast                                       2.756098   
               West                                            3.016129   

                            Ayva Units - 26 Week  Ayva Units - 52 Week  \
2L+ TKI Target Region                                                    
No             Central                  0.005017              0.005017   
               Great Lakes              0.000000              0.000000   
               Northeast                0.000000              0.000000   
               South                    0.000000              0.000000   
               Southeast                0.000000              0.000000   
               West                     0.002584              0.002584   
Yes            Central                  0.026178              0.047120   
               Great Lakes              0.014706              0.014706   
               Northeast                0.000000              0.016043   
               South                    0.041237              0.103093   
               Southeast                0.052133              0.061611   
               West                     0.050179              0.064516   

                            Facility Zip  Field Reported 2L+  \
2L+ TKI Target Region                                          
No             Central      64390.398630            0.000000   
               Great Lakes  34963.549595            0.000000   
               Northeast    10198.255519            0.000000   
               South        70357.611506            0.000000   
               Southeast    31826.834819            0.000000   
               West         92779.757794            0.000000   
Yes            Central      60466.223214            1.489247   
               Great Lakes  34494.000000            1.656805   
               Northeast     9371.661905            1.333333   
               South        71173.211268            1.356688   
               Southeast    31603.441296           

In [19]:
pd.pivot_table(df,index=["2L+ TKI Target","Region"],values=["April Marketing Campaign - Email Open Count"],aggfunc=[np.sum])

sum
                           April Marketing Campaign - Email Open Count
2L+ TKI Target Region                                                 
No             Central                                           885.0
               Great Lakes                                       814.0
               Northeast                                        1033.0
               South                                             710.0
               Southeast                                         824.0
               West                                              671.0
Yes            Central                                           140.0
               Great Lakes                                       171.0
               Northeast                                         132.0
               South                                             150.0
               Southeast                                         113.0
               West                                              187.0

In [22]:
pd.pivot_table(df,index=["2L+ TKI Target","Region"],values=["Total 3L+ GIST Patients","April Marketing Campaign - Email Open Count"],aggfunc=[np.sum])

sum  \
                           April Marketing Campaign - Email Open Count   
2L+ TKI Target Region                                                    
No             Central                                           885.0   
               Great Lakes                                       814.0   
               Northeast                                        1033.0   
               South                                             710.0   
               Southeast                                         824.0   
               West                                              671.0   
Yes            Central                                           140.0   
               Great Lakes                                       171.0   
               Northeast                                         132.0   
               South                                             150.0   
               Southeast                                         113.0   
               West                                              187.0   

                                                    
                           Total 3L+ GIST Patients  
2L+ TKI Target Region                               
No             Central                         6.0  
               Great Lakes                    10.0  
               Northeast                       1.0  
               South                           7.0  
               Southeast                       9.0  
               West                            8.0  
Yes            Central                        31.0  
               Great Lakes                    44.0  
               Northeast                      53.0  
               South                          45.0  
               Southeast                      49.0  
               West                           34.0

In [25]:
df["2L+ TKI Target"].value_counts()

No     9441
Yes    1490
Name: 2L+ TKI Target, dtype: int64

In [31]:
pd.pivot_table(df,index=["2L+ TKI Target","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients"],
               aggfunc=[np.sum])

sum                            
                           Total 3L+ GIST Patients Total Treated GIST Patients
2L+ TKI Target Region                                                         
No             Central                         6.0                       418.0
               Great Lakes                    10.0                       499.0
               Northeast                       1.0                       502.0
               South                           7.0                       530.0
               Southeast                       9.0                       488.0
               West                            8.0                       327.0
Yes            Central                        31.0                       382.0
               Great Lakes                    44.0                       472.0
               Northeast                      53.0                       610.0
               South                          45.0                       431.0
               Southeast                      49.0                       524.0
               West                           34.0                       522.0

+ (41) 3L Patients among 9441 HCPs **THAT ARE NOT** 2L plus
+ (256) 3L Patients among 1490 HCPs **THAT ARE** 2L plus